### PR TITLE
DTSPO-17593 - dynamic pipeline name

### PR DESCRIPTION
**Pull request by cpareek**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DTSPO-17593

### Change description ###

Dynamically set pipeline name
### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines

## 🤖AEP PR SUMMARY🤖

Request to AEP failed to process

### CODE REVIEW

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -251,3 +255,5 @@ stages:
                     test_pattern: apim.test.ts
                   ${{ else }}:
                     test_pattern: pipeline.test.ts
+
+name: "$(finalAction) - ${{ parameters.location }} - SDS Azure Platform"
```
Comment (thomast1906) [inline azure-pipelines.yaml:259]: is location needed?

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -191,6 +190,11 @@ variables:
   - template: vars/input-variables.yaml@cnp-azuredevops-libraries
   - name: timeoutInMinutes
     value: 60
+  - name: finalAction
```
Comment (thomast1906) [inline azure-pipelines.yaml:193]: why not just parameters.action?

Comment (cpareek) [reply azure-pipelines.yaml:193]: so parameters.action  will always print  `plan` as its default value even if its `apply`

File: azure-pipelines.yaml (hunk 1 above)
Comment (cpareek) [reply azure-pipelines.yaml:259]: sure -can remove it, kept it as it is as only want to set the initial action

File: azure-pipelines.yaml [hunk 3]
```diff
@@ -191,6 +190,11 @@ variables:
...
+    ${{ if and(eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true)) }}:
+      value: 'apply'
```
Comment (thomast1906) [inline azure-pipelines.yaml:195]: Is that correct? I could run a plan of main branch

**Review by thomast1906 — APPROVED** [review]
Can cft be the same?

**Comment by cpareek** [comment]
> Can cft be the same?

Will change on CFT too

File: azure-pipelines.yaml (hunk 3 above)
Comment (charlesluokb) [reply azure-pipelines.yaml:195]: do we allow runing a plan of main branch that is not auto triggered?

**Pull request merged**